### ISSUE TITLE
cWS only usable on x86_64 architecture

### ISSUE THREAD

**Issue by aeris** [ClusterWS/cWS#21]
Currently, cWS is provided with pre-built binaries for Linux and Darwin and for some hardcoded Node API. Those binaries are for x86_64 only, and so this project is not usable on ARM CPU (Raspberry Pi, Olimex…) or old x86 hardware.

Those pre-built binaries use also 6.5MB of disk space for 16 supported environment, when only 1 is necessary and take only 500kB.

Why not using [`node-gyp`](https://github.com/nodejs/node-gyp) to build the corresponding binary for the targeted system during npm/yarn setup? :thinking:

**Comment by goriunov** [comment]
@aeris it is automatically tries to re build binaries on install (in `package.json`)

```
"install": "node-gyp rebuild > build_log.txt 2>&1 || exit 0",
```
we need pre-built binaries for systems such as windows which by default do not have c++ build tools will all deps.

**Comment by aeris** [comment]
Seems not. I don't see any build at install and running code after that don't work because trying to use x86_64 binaries.

**Comment by aeris** [comment]
```
$ uname -a
Linux kamino 4.15.11-mainline-rev1 #1 SMP Mon Mar 19 08:29:15 UTC 2018 aarch64 GNU/Linux
$ yarn add @clusterws/cws                 
yarn add v1.17.3
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
success Saved 1 new dependency.
info Direct dependencies
└─ @clusterws/cws@0.15.0
info All dependencies
└─ @clusterws/cws@0.15.0
Done in 6.26s.
$ find -name "cws_linux_*.node" | xargs file
./node_modules/@clusterws/cws/dist/cws_linux_67.node: ELF 64-bit LSB pie executable, x86-64, version 1 (GNU/Linux), dynamically linked, BuildID[sha1]=e1115f22b1c38c8a4390ef52dac6be7717e4aa57, stripped
./node_modules/@clusterws/cws/dist/cws_linux_72.node: ELF 64-bit LSB pie executable, x86-64, version 1 (GNU/Linux), dynamically linked, BuildID[sha1]=3a9eb000c5ab2a2e7348d6f9cc169c298cc64421, stripped
./node_modules/@clusterws/cws/dist/cws_linux_64.node: ELF 64-bit LSB pie executable, x86-64, version 1 (GNU/Linux), dynamically linked, BuildID[sha1]=13366587b2b0a93ee12dcfce4b160bf2c7e1edb2, stripped
./node_modules/@clusterws/cws/dist/cws_linux_59.node: ELF 64-bit LSB pie executable, x86-64, version 1 (GNU/Linux), dynamically linked, BuildID[sha1]=a52d273892920cc9bcbc29d4bb5dcdce4e06cd54, stripped
./node_modules/@clusterws/cws/dist/cws_linux_57.node: ELF 64-bit LSB pie executable, x86-64, version 1 (GNU/Linux), dynamically linked, BuildID[sha1]=cb979ec2681ea1e2f3ee4cfb532d2ecc2a4fa3e7, stripped

**Comment by goriunov** [comment]
Hmm, do you see build_log.txt in `node_modules/@clusterws/cws` folder ?

**Comment by aeris** [comment]
Ok, missing python2…
```
gyp ERR! configure error 
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (/opt/nodeenv/versions/11.15.0/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:484:19)
gyp ERR! stack     at PythonFinder.<anonymous> (/opt/nodeenv/versions/11.15.0/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:406:16)
gyp ERR! stack     at F (/opt/nodeenv/versions/11.15.0/lib/node_modules/npm/node_modules/which/which.js:68:16)
gyp ERR! stack     at E (/opt/nodeenv/versions/11.15.0/lib/node_modules/npm/node_modules/which/which.js:80:29)
gyp ERR! stack     at /opt/nodeenv/versions/11.15.0/lib/node_modules/npm/node_modules/which/which.js:89:16
gyp ERR! stack     at /opt/nodeenv/versions/11.15.0/lib/node_modules/npm/node_modules/isexe/index.js:42:5
gyp ERR! stack     at /opt/nodeenv/versions/11.15.0/lib/node_modules/npm/node_modules/isexe/mode.js:8:5
gyp ERR! stack     at FSReqCallback.oncomplete (fs.js:158:21)
gyp ERR! System Linux 4.15.11-mainline-rev1
gyp ERR! command "/opt/nodeenv/versions/11.15.0/bin/node" "/opt/nodeenv/versions/11.15.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /tmp/test/node_modules/@clusterws/cws
gyp ERR! node -v v11.15.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
```
No error/warning on install when building crash :smile:

**Comment by goriunov** [comment]
I will consider adding warning if post install fails with reference to the file. In most of the cases system will pick up pre-built version and that would not be an issue.

Thank you :)

**Comment by goriunov** [comment]
Released new 0.15.1 version with minor improvements where during importing binding it will attach information about `build_log.txt` file

```
Error: Cannot find module './cws_linux_64' check './node_modules/@clusterws/cws/build_log.txt' for post install build logs
```
